pkg/auth/rbac.go

```diff
@@ -201,11 +203,22 @@ func ReadRbacPolicy(dexEnabled bool) (policy map[string]*Permission, err error)
 
 // Checks user permissions on the RBAC policy.
 func CheckUserPermissions(rbacConfig RBACConfig, user *User, env, envGroup, application, action string) error {
-	permissionsWanted := fmt.Sprintf("%s,%s,%s:%s,%s,allow", user.DexAuthContext.Role, action, envGroup, env, application)
-	_, permissionsExist := rbacConfig.Policy[permissionsWanted]
-	if !permissionsExist {
-		return status.Errorf(codes.PermissionDenied, fmt.Sprintf("user does not have permissions for: %s", permissionsWanted))
+	// If the action is environment independent, the env format is <ENVIRONMENT_GROUP>:*
+	if isEnvironmentIndependent(action) {
+		env = "*"
 	}
-
-	return nil
+	// Check for wildcard application permission.
+	permissionApplicationWildcard := fmt.Sprintf(PermissionTemplate, user.DexAuthContext.Role, action, envGroup, env, "*")
+	_, permissionsExist := rbacConfig.Policy[permissionApplicationWildcard]
+	if permissionsExist {
+		return nil
+	}
+	// Check if the permission exists.
+	permissionsWanted := fmt.Sprintf(PermissionTemplate, user.DexAuthContext.Role, action, envGroup, env, application)
+	_, permissionsExist = rbacConfig.Policy[permissionsWanted]
+	if permissionsExist {
+		return nil
+	}
```

Comment:
In the future, this needs to be a loop over all permutations (env can also be "*" at least)
But it's fine for now.  Keep it.

Reply:
(I am referring to actions with `isEnvironmentIndependent(action) == false`)

Reply:
Yes, for now I kept it simple. I added a TODO for that.